src/main/java/com/wanderersoftherift/wotr/effects/CriticalEffect.java

```diff
@@ -0,0 +1,29 @@
+package com.wanderersoftherift.wotr.effects;
+
+import com.wanderersoftherift.wotr.init.Attributes;
+import net.minecraft.util.RandomSource;
+import net.minecraft.world.entity.LivingEntity;
+
+public class CriticalEffect {
+    public static float calcFinalDamage(
+            float initialDamage,
+            LivingEntity attacker,
+            LivingEntity foe,
+            RandomSource random) {
+
+        int critChance = (int) attacker.getAttributeValue(Attributes.CRITICAL_CHANCE)
+                - (int) foe.getAttributeValue(Attributes.CRITICAL_AVOIDANCE);
+        float critMult = (float) attacker.getAttributeValue(Attributes.CRITICAL_MULTIPLIER);
+
+        int critApplications = 0;
+
+        if (critChance > 100) {
+            critApplications = critChance / 100;
+        } else {
+            if (random.nextInt(0, 100) < (critChance % 100)) {
+                critApplications += 1;
+            }
+        }
```

Comment:
Please, stop being fixated on those ifs

```suggestion
        int critApplications = (critChance + random.nextInt(100)) / 100;
```